src/Datadog.Trace/TraceId.cs

```diff
@@ -91,12 +84,8 @@ public static TraceId CreateRandom()
         /// <returns>Instance of randomly generated <see cref="TraceId"/>.</returns>
         public static TraceId CreateRandomDataDogCompatible()
         {
-            var lowerBytes = new byte[8];
-
             var randomNumberGenerator = RandomNumberGenerator.Current;
-            Unsafe.WriteUnaligned(ref lowerBytes[0], randomNumberGenerator.Next());
-
-            var lower = (ulong)BitConverter.ToInt64(lowerBytes, startIndex: 0) & 0x7FFFFFFFFFFFFFFF;
+            var lower = (ulong)randomNumberGenerator.Next() & 0x7FFFFFFFFFFFFFFF;
```

Comment:
Side comment:
AFAIK `ulong` is not [CLS Compliant](https://docs.microsoft.com/en-us/dotnet/api/system.clscompliantattribute?view=net-5.0). Is it not an issue? Shouldn't we add `[assembly: CLSCompliant(true)]` to the Tracer?

Reply:
Should we create issue for that? We would have to change spanId to be long as well, as for now we are using ulong for that as well.

Reply:
Done https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/issues/131